sources/packages/backend/apps/api/src/route-controllers/program-year/program-year.institutions.controller.ts

```diff
@@ -0,0 +1,38 @@
+import { Controller, Get, NotFoundException } from "@nestjs/common";
+import { ProgramYearService } from "../../services";
+import { ClientTypeBaseRoute } from "../../types";
+import BaseController from "../BaseController";
+import { AllowAuthorizedParty } from "../../auth/decorators/authorized-party.decorator";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { OptionItemAPIOutDTO } from "../models/common.dto";
+import { IsBCPublicInstitution } from "../../auth/decorators";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("program-year")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-program-year`)
+export class ProgramYearInstitutionsController extends BaseController {
+  constructor(private readonly programYearService: ProgramYearService) {
+    super();
+  }
+
+  /**
+   * Gets a list of program years returned as option items (id/description pair).
+   * @returns an array of program years as id/description objects.
+   */
+  @IsBCPublicInstitution()
+  @Get("options-list")
+  @ApiNotFoundResponse({
+    description: "No program years were found.",
+  })
+  async getProgramYears(): Promise<OptionItemAPIOutDTO[]> {
+    const programYears = await this.programYearService.getProgramYears();
```

Comment:
Outside the scope of the PR by too close because the method used does not ensure the correct order.
Can you please replace the current implementation of the `getProgramYears` with the below one?
```ts
/***
   * Get all active the program years information.
   * @returns active program years ordered by name in descendent order.
   */
  async getProgramYears(): Promise<ProgramYear[]> {
    return this.repo.find({
      select: {
        programYear: true,
        programYearDesc: true,
      },
      where: { active: true },
      order: { programYear: "DESC" },
    });
  }
```